flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesClusterDescriptor.java

```diff
@@ -240,20 +238,11 @@ private String getWebMonitorAddress(Configuration configuration) throws Exceptio
 
         flinkConfig.setString(KubernetesConfigOptionsInternal.ENTRY_POINT_CLASS, entryPoint);
 
-        // Rpc, blob, rest, taskManagerRpc ports need to be exposed, so update them to fixed values.
-        KubernetesUtils.checkAndUpdatePortConfigOption(
-                flinkConfig, BlobServerOptions.PORT, Constants.BLOB_SERVER_PORT);
-        KubernetesUtils.checkAndUpdatePortConfigOption(
-                flinkConfig, TaskManagerOptions.RPC_PORT, Constants.TASK_MANAGER_RPC_PORT);
-        KubernetesUtils.checkAndUpdatePortConfigOption(
-                flinkConfig, RestOptions.BIND_PORT, Constants.REST_PORT);
+        KubernetesUtils.updatePorts(flinkConfig);
```

Comment:
It seems that the old changes have been brought back again.

Reply:
squash and merge commit caused some errors...

Reply:
Hmm. Could you please get this resolved carefully? I believe this PR is almost ready for merge.

Reply:
The code seems to be organized and tested on a real k8s cluster